libs/traces/src/main/java/de/westnordost/osmapi/traces/GpsTracesApi.java

```diff
@@ -79,6 +84,59 @@ protected void write() throws IOException
 		
 		return osm.makeAuthenticatedRequest(GPX + "/create", "POST", writer, new IdResponseReader());
 	}
+
+	/**
+	 * Upload a new trace from GPX file.
+	 *
+	 * @param name this is usually the "file name" of the GPX trace
+	 * @param visibility the visibility the trace should have
+	 * @param description short description of the trace. May not be null or empty.
+	 * @param tags keywords with which this trace can be found. May be null.
+	 * @param gpxFile The File to upload
+	 * @return trace id
+	 *
+	 * @throws IllegalArgumentException if either name, description or any single tag is longer than
+	 *                                  255 characters
+	 * @throws OsmBadUserInputException if the trace is invalid
+	 * @throws OsmAuthorizationException if this application is not authorized to write traces
+	 *                                   (Permission.WRITE_GPS_TRACES)
+	 */
+	public long create(
+			final String name, final GpsTraceDetails.Visibility visibility,
+			final String description, final List<String> tags,
+			final File gpxFile)
```

Comment:
As noted, wouldn't it be more flexible if this accepted an InputStream rather than a File?